### ISSUE TITLE
Process limit across 2 machines not being followed

### ISSUE THREAD

**Issue by butsjoh** [deanpcmad/sidekiq-limit_fetch#74]
This might be related to https://github.com/brainopia/sidekiq-limit_fetch/issues/56

Again we experience some weird issue with our sidekiq configuration. We currently have 2 machines which have the same sidekiq configuration:

```

---
:verbose: true
:concurrency: 10
:tag: sidekiq_disk
:queues:
- - videos
  - 7
:process_limits:
  videos: 1
```

Whenever we queue jobs in sidekiq it seems that one machine sticks to the limit of only processing one at the time but the other machine is not. No clue what is going on here and we have a deploy script which provisions the machine equally.

![screen shot 2016-06-30 at 12 51 15](https://cloud.githubusercontent.com/assets/96848/16486002/451d5942-3ec2-11e6-8044-43710a5b09c7.png)

When asking the queue programmatically what is is doing we get the following output:

Machine 1:

```
"Current sidekiq process: e1cffa43-3334-4888-8153-4426dcc2c3b5\n\n  All processes:\nd50199d2-6039-4c98-b7ea-f9a9583a7a54\n099f32b2-aded-4b6f-b5dc-f4ebfde0acbb\na3a54913-293e-456c-9a33-76e03777e9cd\nda0d15d7-1724-4406-ace3-d0b8ad016cc0\nf0dca400-0cde-498a-a30e-13597d815e65\n\n  Stale processes:\n\n\n  Locked queue processes:\n099f32b2-aded-4b6f-b5dc-f4ebfde0acbb\n\n  Busy queue processes:\n099f32b2-aded-4b6f-b5dc-f4ebfde0acbb\n\n  Limit:\nnil\n\n  Process limit:\n1\n\n  Blocking:\n\n"
```

Machine 2:

```
 "Current sidekiq process: c7259ef0-99eb-4688-8d2f-4c244b7100b8\n\n  All processes:\nd50199d2-6039-4c98-b7ea-f9a9583a7a54\n099f32b2-aded-4b6f-b5dc-f4ebfde0acbb\na3a54913-293e-456c-9a33-76e03777e9cd\nda0d15d7-1724-4406-ace3-d0b8ad016cc0\nf0dca400-0cde-498a-a30e-13597d815e65\n\n  Stale processes:\n\n\n  Locked queue processes:\n099f32b2-aded-4b6f-b5dc-f4ebfde0acbb\n56470ac6-6909-4c20-8512-de1aaa3f780f\n\n  Busy queue processes:\n099f32b2-aded-4b6f-b5dc-f4ebfde0acbb\n\n  Limit:\nnil\n\n  Process limit:\n1\n\n  Blocking:\n\n"
```

No clue why this is happening ...

**Comment by butsjoh** [comment]
These are our sidekiq deps:

<img width="325" alt="screen shot 2016-06-30 at 12 51 37" src="https://cloud.githubusercontent.com/assets/96848/16486020/592f2942-3ec2-11e6-9fa4-db67e4403f21.png">

**Comment by janiskesteris** [comment]
Have the same or similar issue:
my sidekiq.yml
```yml
production:
  :concurrency: 15
:queues:
  - ["memory_intensive_job", 1]
  - ["job2", 1]
  - ["job3", 1]
:process_limits:
  memory_intensive_job: 1
  job2: 10
  job3: 10
```

- 10 sidekiq worker machines 1 server for populating the queues with 1 redis server
- each worker machine has 3 processes each with 15 concurrency as per sidekiq.yml
- memory_intensive_job is consumes a lot of memory so limit per process is set to 1

Expectation:
10 machines * 3 processes * 1 limit for memory_intensive_job = 30 concurrent jobs being executed

Reality:
starts with 30 but over time process takes more and more memory_intensive_jobs which eventually ends up in worker server crashing and not being responsive.

**Comment by Yoshyn** [comment]
Hi,

We also had such concerns.
We no longer have them without knowing why.

But. You can take a look on this issue : https://github.com/brainopia/sidekiq-limit_fetch/issues/84

I wrote a demo to see if there's a concern with process_limits. 
Unfortunately (or not), I found nothing. You may be luckier. In any case this can give you a basis for testing a minimal case.

**Comment by stevestofiel** [comment]
We had a very similar issue.  Multiple machines reading off of the same queue but each machine needed to be limited to processing one job at a time from that queue.

The way we read the documentation it seemed that we needed to use process limits

```yaml
:process_limits:
  single_resoruce_queue: 1
```
 but I am now thinking this only applies if you are running multiple sidekiq processes on the same machine.

We added
```yaml
:limits:
  single_resoruce_queue: 1
```
and are seeing drastically reduced collisions/issues.

Now multiple machines are pulling off the same queue and each machine is only processing 1 job at a time. (of course we have many other queues that use concurrency)
